pkg/client/testclient/fixture.go

```diff
@@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
 
 	return nil
 }
+
+func (o objects) Delete(obj runtime.Object) error {
```

Comment:
That's just not how this is set up to work.  You describe sequences of return values, rather than manipulating the store.  Tests that call delete are given the response for the kind, and you can set up an ordered list of response types for a given call.

----- Original Message -----
> > @@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
> >  
> >  	return nil
> >  }
> > +
> > +func (o objects) Delete(obj runtime.Object) error {
> 
> How does this break clients or other tests?  I don't understand.  Tests must
> call Delete specifically.
> 
> For my test, I would like to have a PV and PVClaim for my test, ensure they
> bind correctly, then delete the PVClaim to test that the volume releases.
> Because I use the client in my binder, removing the claim from the test
> client is the easiest and most natural way to mimic the user deleting the
> claim.
> 
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6928/files#r28541967